postgres/src/main/java/org/jdbi/v3/postgres/JsonArgumentFactory.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.postgres;
+
+import java.lang.reflect.Type;
+import java.sql.Types;
+import java.util.Optional;
+
+import org.jdbi.v3.core.argument.Argument;
+import org.jdbi.v3.core.argument.ArgumentFactory;
+import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.json.Json;
+
+class JsonArgumentFactory implements ArgumentFactory {
+    @Override
+    public Optional<Argument> build(Type type, Object value, ConfigRegistry config) {
+        if (type != Json.class) {
```

Comment:
What's happening here? `Json.class` isn't the type, is the qualifier. Shouldn't this be an annotation on the factory class declaration?

Reply:
This is the "poor man's SPI" to decouple Postgres support from Json support.
The Json binder matches parameterized types e.g. `@Json Foo`, makes them a Json string, then re-binds them as the type `Json.class` (bit of a hack, but it works!)

It expects the database engine, in this case Postgres, to provide the specific binding strategy.

I guess this could be an ArgumentFactory setting on the config instead if that's clearer.  But we already have an argument factory registry...

Reply:
It sounds like `QualifiedArgumentFactory` wants to be API.